### ISSUE TITLE
`dotnet pack` product broken .nupkg with the latest daily build(2.2.0-preview1)

### ISSUE THREAD

**Issue by livarcocc** [dotnet/corefx#25803]
_From @yyjdelete on December 7, 2017 1:34_

Test with dotnet sdk `2.2.0-preview1-007736`/7622/7460, Windows 10 x64(17064).
And `2.1.1` in VS15.5preview5 works well.

# Step:
1. Install the latest daily build of sdk
2. `dotnet new classlib`
3. `dotnet pack`
4. Open the output .nupkg with `NuGet Package Explorer` or some zip manager( and try unpack it), and get an error.
5. Found the `.nuspec` file in the `nupkg` has wrong CRC(but the packed data is right)

_Copied from original issue: dotnet/sdk#1801_

**Comment by livarcocc** [comment]
This issue was moved to NuGet/Home#6267

**Comment by livarcocc** [comment]
Turns out that this is potentially an issue in the framework. Moving it to corefx.

**Comment by livarcocc** [comment]
```
using System;
using System.IO;
using System.IO.Compression;
using System.Xml.Linq;

namespace Test
{
    class Program
    {
        static void Main(string[] args)
        {
            using (var fs = File.Create("test.zip"))
            using (var package = new ZipArchive(fs, ZipArchiveMode.Create, leaveOpen: true))
            {
                //This file has an bad crc in netcore2.1
                ZipArchiveEntry entry1 = package.CreateEntry("x1.txt", CompressionLevel.Optimal);
                using (Stream stream = entry1.Open())
                {
                    CreateDocument().Save(stream);
                    //Console.WriteLine(stream.Position);//50
                }

                ZipArchiveEntry entry2 = package.CreateEntry("x2.txt", CompressionLevel.Optimal);
                using (Stream stream = entry2.Open())
                using (var writer = new StreamWriter(stream, Encoding.UTF8))//make both has an BOM
                {
                    CreateDocument().Save(writer);
                    writer.Flush();
                    //Console.WriteLine(stream.Position);//50
                }
            }
        }

        public static XDocument CreateDocument()
        {
            return new XDocument(
                new XElement("Foo")
                );
        }
    }
}
```

The program above can repro it. Copied from @yyjdelete

**Comment by joshfree** [comment]
@ianhays could you take a look at this for 2.1?
/cc @JeremyKuhne

**Comment by ianhays** [comment]
This may be related to https://github.com/dotnet/corefx/issues/25665 which came in https://github.com/dotnet/corefx/pull/25624. That's the only change that occurred in the past month that might cause the above failure. @livarcocc what build were you using before this that didn't have this bug?

cc: @ViktorHofer

**Comment by JeremyKuhne** [comment]
Note that  #25665 is referencing the fact that we removed the check for invalid path characters. While it is possible that caused this, it doesn't seem likely.

cc: @Anipik

**Comment by yyjdelete** [comment]
@ianhays 
Tested with build from https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.NETCore.App
`2.1.0-preview2-25430-06` works well but `2.1.0-preview2-25615-02` not.

**Comment by ianhays** [comment]
Good catch @yyjdelete. The pointer declaration is now `fixed (byte* bufferPtr = buffer)` but used to be `fixed (byte* buf = &buffer[offset])` before https://github.com/dotnet/corefx/pull/22927.

**Comment by yyjdelete** [comment]
@ianhays Should this also be closed by #25899 ?

**Comment by ianhays** [comment]
Yes it should, thanks @yyjdelete!
